app/scripts/components/analysis/results/use-analysis-params.ts

```diff
@@ -130,3 +130,18 @@ export function useAnalysisParams(): AnalysisParams {
 
   return params;
 }
+
+export function analysisParams2QueryString(
+  params: Omit<AnalysisParams, 'errors'>
+) {
+  const urlParams = qs.stringify({
+    start: params.date.start ? userTzDate2utcString(params.date.start) : undefined,
+    end: params.date.end ? userTzDate2utcString(params.date.end) : undefined,
+    datasets: params.datasetsLayers
+      ? params.datasetsLayers.map((d) => d.id).join('|')
+      : undefined,
+    aoi: params.aoi ? polygonUrlEncode(params.aoi, 4) : undefined
+  });
```

Comment:
Also flattening allows simplifying the `AnalysisParams` type, [see](https://github.com/NASA-IMPACT/delta-ui/pull/284/commits/772896aa26f82ae049d554b432ef2a68254c5e2d#diff-596a09dbcd5b51c32ef05f6e75f0bfce13b7ab39e11f87e05935a4e158014ce7R20)

Reply:
One more thing: makes it possible to generically write to the state, [see](https://github.com/NASA-IMPACT/delta-ui/pull/284/commits/5ccc619bc5c51ed35c74b5a844bd8a53aa382ba0#diff-596a09dbcd5b51c32ef05f6e75f0bfce13b7ab39e11f87e05935a4e158014ce7R133-R136)

Reply:
For some reason I like having the related dates together, but I agree that flattening helps.
Not changing it here, since you already did quite some work on your branch. Let's do the rest there as well.